src/smart-components/access-management/UserGroupsTable.tsx

```diff
@@ -162,25 +190,79 @@ const UserGroupsTable: React.FunctionComponent<UserGroupsTableProps> = ({
-            pageSelected={pageSelected}
-            pagePartiallySelected={pagePartiallySelected}
-            onSelect={handleBulkSelect}
+    <Fragment>
+      {isDeleteModalOpen && (
+        <WarningModal
+          ouiaId={`${ouiaId}-remove-user-modal`}
+          isOpen={isDeleteModalOpen}
+          title={
+            <FormattedMessage
+              {...messages.deleteUserGroupModalTitle}
+              values={{
+                count: currentGroups.length,
+                plural: currentGroups.length > 1 ? intl.formatMessage(messages.groups) : intl.formatMessage(messages.group),
+              }}
+            />
+          }
+          withCheckbox
+          checkboxLabel={intl.formatMessage(messages.understandActionIrreversible)}
+          confirmButtonLabel={intl.formatMessage(messages.remove)}
+          confirmButtonVariant={ButtonVariant.danger}
+          onClose={() => setIsDeleteModalOpen(false)}
+          onConfirm={() => {
+            handleDeleteGroups(currentGroups);
+          }}
+        >
+          <FormattedMessage
+            {...messages.deleteUserGroupModalBody}
+            values={{
+              b: (text) => <b>{text}</b>,
+              count: currentGroups.length,
+              plural: currentGroups.length > 1 ? intl.formatMessage(messages.groups) : intl.formatMessage(messages.group),
+              name: currentGroups[0]?.name,
+            }}
           />
-        }
-        pagination={React.cloneElement(paginationComponent, { isCompact: true })}
-      />
-      <DataViewTable variant="compact" aria-label="Users Table" ouiaId={`${ouiaId}-table`} columns={COLUMNS} rows={rows} />
-      <DataViewToolbar ouiaId={`${ouiaId}-footer-toolbar`} pagination={paginationComponent} />
-    </DataView>
+        </WarningModal>
+      )}
+      <DataView ouiaId={ouiaId} selection={selection}>
+        <DataViewToolbar
+          ouiaId={`${ouiaId}-header-toolbar`}
+          bulkSelect={
+            <BulkSelect
+              isDataPaginated
+              pageCount={groups.length}
+              selectedCount={selected.length}
+              totalCount={totalCount}
+              pageSelected={pageSelected}
+              pagePartiallySelected={pagePartiallySelected}
+              onSelect={handleBulkSelect}
+            />
+          }
+          actions={
+            <div data-ouia-component-id={`${ouiaId}-actions-dropdown`}>
+              <ActionsColumn
+                items={[
+                  {
+                    title: intl.formatMessage(messages.usersAndUserGroupsEditUserGroup),
+                    onClick: () => console.log('EDIT USER GROUP'),
+                  },
+                  {
+                    title: intl.formatMessage(messages.usersAndUserGroupsDeleteUserGroup),
+                    onClick: () => {
+                      handleDeleteModalToggle(groups.filter((group) => selected.some((selectedGroup) => selectedGroup.id === group.uuid)));
+                    },
```

Comment:
`selectedGroup` has implicit `any` type
```suggestion
                    onClick: () => {
                      handleDeleteModalToggle(
                        groups.filter((group) => selected.some((selectedGroup: { id: string }) => selectedGroup.id === group.uuid))
                      );
                    },
```